plugins/destination/gcs/client/read.go

```diff
@@ -31,17 +18,5 @@ func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName strin
 	}
 	defer r.Close()
 
-	switch c.pluginSpec.Format {
-	case FormatTypeCSV:
-		if err := csv.Read(r, table, sourceName, res); err != nil {
-			return err
-		}
-	case FormatTypeJSON:
-		if err := json.Read(r, table, sourceName, res); err != nil {
-			return err
-		}
-	default:
-		panic("unknown format " + c.pluginSpec.Format)
-	}
-	return nil
+	return c.Client.Read(r, table, sourceName, res)
```

Comment:
rm this func (use embedded). possibly move noRotate check to embedded client

Reply:
Read has a different function signature...